src/asr_got_reimagined/domain/stages/stage_8_reflection.py

```
@@ -25,432 +31,239 @@ class AuditCheckResult(BaseModel):
+            falsifiable_count = sum(1 for r in results if r["has_criteria"])
+            ratio = falsifiable_count / total_hypotheses if total_hypotheses else 0
+            message = f"{falsifiable_count}/{total_hypotheses} ({ratio:.2%}) hypotheses have falsifiability criteria."
+            status = "PASS" if ratio >= self.min_falsifiable_hypothesis_ratio else ("WARNING" if ratio > 0 else "FAIL")
+            return AuditCheckResult(check_name="hypothesis_falsifiability", status=status, message=message)
+        except Neo4jError as e:
+            logger.error(f"Neo4j error in falsifiability check: {e}")
+            return AuditCheckResult(check_name="hypothesis_falsifiability", status="FAIL", message=f"Query error: {e}")
+
+    async def _check_statistical_rigor_from_neo4j(self) -> AuditCheckResult:
+        query = """
+        MATCH (e:Node:EVIDENCE) 
```

Ruff (0.11.9)

Trailing whitespace

Remove trailing whitespace

(W291)